### DISCUSSION TITLE
Data migration tool for DB normalisation via App

### DISCUSSION THREAD

**Discussion by VaZark** [directus/directus#11635] [Feature Requests]
As we build and enable more nocode / low code solutions, sometimes junior devs or business designs DB thats are not particularly normalises or efficient. (Speaking of experience)

It’d be an amazing feature if we can “refactor” collections via UI. It’ll also be a significant highlight for directus vis-à-vis similar projects.

1. Select fields from existing collections to be moved to a separate collection.(autogenerate foreign key)
2. Create the new table with FK
3. Copy the data and validate it
4. Drop columns in original collection.

While each of these steps are manually possible, a nocode way of doing it via App would interesting to have.

**Comment by xu4wang** [comment]
This proposal may also help the use case in #11612

It will be great if the user can sync one database(dev) to another(production) via the APP.

**Comment by VaZark** [comment]
Is there any way to extend and add additional sections in the data-model view or should I just go ahead with a new module for now? I'm trying to give a shot at implementing this myself

**Reply by rijkvanzanten** [reply]
There's no way to inject custom front-end into the existing settings module, so a custom module would be the way to go here 🙂

**Comment by rijkvanzanten** [comment]
I'm wondering what the 80/20 is for a feature like this in the app itself. To me this sounds like a feature that would only be used once or twice by the administrators who know what they're doing in the database, which might be better suited as a CLI command for example 🤔

**Reply by VaZark** [reply]
Since it’s just meant to help maintaining the DBs created via nocode easier rather than opening  the actual db, I’d assume almost everyone would use it but they will probably use it rarely.

So, in all honesty,  I’m not sure about how to evaluate the 80/20 for that  case.

**Reply by rijkvanzanten** [reply]
> I’d assume almost everyone would use it

Hmm that's the thing I'm wondering about. Just from my own limited personal experience, I haven't heard anybody else discuss this, nor have had the need for an automated cleanup from Directus' side myself 🤔 That makes me wonder if this is something that 80%+ of our user base would actively use, or if it's too much of a power-user / edge-case type thing to add to the core codebase 🤔

**Reply by VaZark** [reply]
I understand that completely. It’s one of those things that is caught too late and is always pushed to be a part of a major refactoring effort and is never done lightly.

The case I’m making is, if people know a feature like this exists, they’ll all use it.

**Comment by rijkvanzanten** [comment]
Heya! Thank you for taking the time to submit this request! 

It has been over **90 days**, and this discussion has not received at least **15 votes** from the community. This means that we don't feel like there's enough community interest to warrant further R&D into this topic at this time. :ice_cube:

This request will now be closed to keep our discussions tidy. Please reach out if you have any questions!

_For more information, see our [Feature Request Process](https://docs.directus.io/contributing/feature-request-process.html)._
